nix/eval-machine-info.nix

```diff
@@ -39,13 +39,42 @@ rec {
         # Get the configuration of this machine from each network
         # expression, attaching _file attributes so the NixOS module
         # system can give sensible error messages.
+
         modules =
           concatMap (n: optional (hasAttr machineName n)
             { imports = [(getAttr machineName n)]; inherit (n) _file; })
           networks;
-      in
-      { name = machineName;
-        value = import <nixpkgs/nixos/lib/eval-config.nix> {
+
+        machineConfs =
+          concatMap (n: optional (hasAttr machineName n)
+            (getAttr machineName n))
+          networks;
+
+        nameToPath = attrs: name: {
+          prefix = name;
+          path = attrs."${name}";
+        };
+
+        attrsetToPaths = attrset: map (nameToPath attrset)
+          (builtins.attrNames attrset);
+
+        importSources =
+          (concatMap (module:
+              if (!builtins.isFunction module
```

Comment:
See https://github.com/NixOS/nixops/pull/665#issuecomment-301757938: we can't introspect a function in this way.